dali/operators/generic/one_hot.h

```diff
@@ -57,7 +57,20 @@ class OneHot : public Operator<CPUBackend> {
         axis_(spec.GetArgument<int>("axis")),
         output_type_(spec.GetArgument<DALIDataType>(arg_names::kDtype)),
         on_value_(spec.GetArgument<float>("on_value")),
-        off_value_(spec.GetArgument<float>("off_value")) {}
+        off_value_(spec.GetArgument<float>("off_value")),
+        layout_axis_name_(spec.GetArgument<std::string>("layout_axis_name")) {
+    auto axis_name_len = layout_axis_name_.length();
+    unsigned char ch;
+    if (axis_name_len > 1 ||
+        (axis_name_len == 1 &&
+         (!std::isupper(ch = static_cast<unsigned char>(layout_axis_name_[0])) &&
+          !std::isdigit(ch)))) {
```

Comment:
These constraints are excessive.

Reply:
What about std::isalnum so that '\n' or '\r' are rejected? Or it's not possible to pass them in the first place?

Reply:
I'd be more worried by non-ASCII characters than by non-alphanumeric. But so far we haven't seen any problems and it's not validated anywhere. The only thing that would wreak havoc is `\0`, but we don't guard against it, either - it would likely cause corrupted layout at some point, but nothing more severe (it wouldn't crash or violate memory access).

Reply:
We have a layout check on the executor level AFAIK. We can add something like this there as well. But just an idea for a separate PR.